src/add-users/add-user.tsx

```diff
@@ -1,39 +1,97 @@
+import { useMutation } from '@apollo/client';
 import React, { useState } from 'react';
-import { SafeAreaView, StatusBar, ScrollView, View, Text, TextInput, Button, useColorScheme } from 'react-native';
+import {
+  SafeAreaView,
+  StatusBar,
+  ScrollView,
+  View,
+  Text,
+  TextInput,
+  Button,
+  useColorScheme,
+  Alert,
+  ActivityIndicator,
+} from 'react-native';
+import { Navigation, NavigationComponentProps } from 'react-native-navigation';
 import { Colors } from 'react-native/Libraries/NewAppScreen';
+import { AddUserMutationDataResponse, createUserMutation, ErrorResponse } from '../apollo/apollo-model';
 import { commonStyles } from '../common/common-style';
 import {
   emailPattern,
-  idPattern,
   namePattern,
+  passwordPattern,
   phonePattern,
   rolePattern,
 } from '../utils/add-user-fields-regex-validation';
 import { addUserFieldsValidation } from '../utils/add-user-fields-validation';
 import { validateDate } from '../utils/date-validation';
 
-export const AddUser = () => {
+export const AddUser = (props: NavigationComponentProps) => {
   const isDarkMode = useColorScheme() === 'dark';
   const backgroundStyle = {
     backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
   };
 
-  const [id, onChangeId] = useState('');
   const [name, onChangeName] = useState('');
   const [phone, onChangePhone] = useState('');
   const [birthDate, onChangeBirthDate] = useState('');
   const [email, onChangeEmail] = useState('');
   const [role, onChangeRole] = useState('');
+  const [password, onChangePassword] = useState('');
+
+  const [createUser, { loading }] = useMutation(createUserMutation);
+  const birthDateSplitted = birthDate.split('/');
+  const birthDateFormatted = birthDateSplitted[2] + '-' + birthDateSplitted[1] + '-' + birthDateSplitted[0];
+  const createUserData = {
+    variables: {
+      data: {
+        name: name,
+        email: email,
+        phone: phone,
+        birthDate: birthDateFormatted,
+        password: password,
+        role: role,
```

Comment:
uma coisa interessante é que quando a variável tem o mesmo nome da propriedade vc n precisa passar ela como valor:
```suggestion
        name,
        email,
        phone,
        birthDate: birthDateFormatted,
        password,
        role,
```